src/Chapter3.hs

```diff
@@ -971,6 +1097,23 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data DayOfWeek = Mon | Tue | Wed | Thu | Fri | Sat | Sun deriving (Show, Eq, Ord, Bounded, Enum)
+
+isWeekend :: DayOfWeek -> Bool
+isWeekend Sat = True
+isWeekend Sun = True
+isWeekend _   = False
+
+nextDay :: DayOfWeek -> DayOfWeek
+nextDay day
+    | day == maxBound = minBound
+    | otherwise       = succ day
+
+daysToParty :: DayOfWeek -> Int
+daysToParty day
+    | fromEnum day < fromEnum Fri = fromEnum Fri - fromEnum day
+    | otherwise                   = 7 - (fromEnum day - fromEnum Fri)
```

Comment:
If you remember, how `mod` works for negative numbers, you can implement this function in a slightly more elegant way :slightly_smiling_face: